src/protocol/input-method-unstable-v2.xml

```diff
@@ -0,0 +1,404 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<protocol name="input_method_unstable_v2">
```

Comment:
Where does this protocol definition come from? As far as I can tell, `wayland-protocols` has only v1 and doesn't even have a PR for adding v2?

We probably want to have a canonical source for this :)

Reply:
Could *we* drive a `wayland-protocols` PR?

Reply:
There's some discussion of this protocol version in the Mir PR https://github.com/MirServer/mir/pull/2139#issuecomment-897971091

And, yes, we ought to be able to drive a `wayland-protocols` PR.

Reply:
Whoops, these are the still the old versions that Squeekboard had. The first step is making sure [Squeekboard matches us and wlroots](https://gitlab.gnome.org/World/Phosh/squeekboard/-/merge_requests/472). After that we can talk to both of those teams to see what standard place these belong in (wayland-protocols or wlr-protocols). Not even Layer Shell is in wayland-protocols (it's been in [PR](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/28) for over a year, so I'm not super enthused about trying to get something in it.

Reply:
I don't think we need to care *too* much about *landing* a PR in wayland-protocols (although that would obviously be ideal). I *do* think we need to care about there being a single, canonical place to get the protocol from. There are already at least two slightly different versions of this protocol around!

Since this is a v2 to a `wp_` protocol already in wayland-protocols, wayland-protocols is the only sensible place to put it.

Reply:
There is extensive discussion of the whole ecosystem in [these](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/issues/39) [two](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/issues/40) `wayland-protocols` issues. The link they give for input-method-v2 is to the wlroots source tree (the same place I got it).